website/teacher.py

```diff
@@ -317,6 +319,97 @@ def remove_invite(user):
         DATABASE.remove_class_invite(username)
         return {}, 200
 
+    @app.route('/for-teachers/create-accounts', methods=['GET'])
+    @requires_login
+    def create_accounts(user):
+        if not is_teacher(user):
+            return 'Only teachers can create multiple accounts', 403
+        classes = DATABASE.get_teacher_classes(user['username'], False)
+
+        return render_template('create-accounts.html', classes=classes)
+
+    @app.route('/for-teachers/create-accounts', methods=['POST'])
+    @requires_login
+    def store_accounts(user):
+        if not is_teacher(user):
+            return 'Only teachers can create multiple accounts', 403
+        body = request.json
+        #Validations
+        if not isinstance(body, dict):
+            return g.auth_texts.get('ajax_error'), 400
+        if not isinstance(body.get('accounts'), list):
+            return "accounts should be a list!", 400
+
+        usernames = []
+        mails = []
+
+        # Validation for correct types and duplicates
+        for account in body.get('accounts', []):
+            if not isinstance(account.get('username'), str):
+                return g.auth_texts.get('username_invalid'), 400
+            if not isinstance(account.get('mail'), str) or not utils.valid_email(account.get('mail')):
+                return g.auth_texts.get('email_invalid'), 400
+            if not isinstance(account.get('password'), str):
+                return g.auth_texts.get('password_invalid'), 400
+            if len(account.get('password')) < 6:
+                return g.auth_texts.get('passwords_six'), 400
+
+            if account.get('username').strip().lower() in usernames:
+                return {'error': g.auth_texts.get('unique_usernames'), 'value': account.get('username')}, 200
+            usernames.append(account.get('username').strip().lower())
+            if account.get('mail').strip().lower() in mails:
+                return {'error': g.auth_texts.get('unique_emails'), 'value': account.get('mail')}, 200
+            mails.append(account.get('mail').strip().lower())
+
+        # Validation for duplicates in the db
+        classes = DATABASE.get_teacher_classes(user['username'], False)
+        for account in body.get('accounts', []):
+            if account.get('class') and account['class'] not in [i.get('name') for i in classes]:
+                return "not your class", 400
```

Comment:
Nitpicky; it would be a bit more consistent to give a 404 here indicating the "class does not exist"

Reply:
Agree, will fix